### ISSUE TITLE
Tailwind @apply not working on libs

### ISSUE THREAD

**Issue by med-albert** [nrwl/nx#7040]
## Current Behavior
@apply utility works on `apps` level but not on `libs`

## Expected Behavior
Should be able to use tailwind's @apply utility.

## Steps to Reproduce
https://github.com/med-albert/tailwind-lib-test

### Environment
```Node : 12.22.6
OS   : darwin x64
yarn : 1.22.11

nx : 12.9.0
@nrwl/angular : Not Found
@nrwl/cli : 12.9.0
@nrwl/cypress : 12.9.0
@nrwl/devkit : 12.9.0
@nrwl/eslint-plugin-nx : 12.9.0
@nrwl/express : Not Found
@nrwl/jest : 12.9.0
@nrwl/linter : 12.9.0
@nrwl/nest : Not Found
@nrwl/next : Not Found
@nrwl/node : Not Found
@nrwl/nx-cloud : Not Found
@nrwl/react : 12.9.0
@nrwl/schematics : Not Found
@nrwl/tao : 12.9.0
@nrwl/web : 12.9.0
@nrwl/workspace : 12.9.0
@nrwl/storybook : 12.9.0
@nrwl/gatsby : Not Found
typescript : 4.3.5
```

**Comment by marckassay** [comment]
Fyi, on this [branch ](https://github.com/marckassay/nunx/tree/feature/add-postcss-config)of my repo, it's able to do what you observed it isn't doing. I don't know if this is recommended way of going about it or not since I haven't received any feedback from an earlier post a few weeks ago.

**Comment by med-albert** [comment]
> Fyi, on this [branch ](https://github.com/marckassay/nunx/tree/feature/add-postcss-config)of my repo, it's able to do what you observed it isn't doing. I don't know if this is recommended way of going about it or not since I haven't received any feedback from an earlier post a few weeks ago.

Hi @marckassay have you tried it with `jit` enabled?

**Comment by med-albert** [comment]
> Fyi, on this [branch ](https://github.com/marckassay/nunx/tree/feature/add-postcss-config)of my repo, it's able to do what you observed it isn't doing. I don't know if this is recommended way of going about it or not since I haven't received any feedback from an earlier post a few weeks ago.

Ah, I saw that you've added a postcss and tailwindconfig on the lib itself. I actually already tried that and it works on development `nx serve` but fails to on build. That's why I removed it and open this issue.

**Comment by marckassay** [comment]
"_... have you tried it with jit enabled?_" I don't believe so. Versus AOT? I never considered if React has that option, as I'm fairly new to React.

**Comment by med-albert** [comment]
Upon further inspection looks like `postcss` is not reaching the `libs` directory,

I tried adding a plugin (`postcss-nesting`) on it and it works on the `apps` folder but not on an imported `lib`.

**Comment by danr-za** [comment]
I think this behavior is as intended.
In our project we have a tailwind [preset](https://tailwindcss.com/docs/presets) in the root and every lib/app has a tailwind config which uses the preset and an additional postcss config respectively.
You can take a look here - https://blog.nrwl.io/setup-next-js-to-use-tailwind-with-nx-849b7e21d8d0

**Comment by med-albert** [comment]
> I think this behavior is as intended.
> In our project we have a tailwind [preset](https://tailwindcss.com/docs/presets) in the root and every lib/app has a tailwind config which uses the preset and an additional postcss config respectively.
> You can take a look here - https://blog.nrwl.io/setup-next-js-to-use-tailwind-with-nx-849b7e21d8d0

@danr-za I don't think so, there's no tailwind/postcss config on the initial tutorial and the reason they added one is because of storybook for testing component in isolation.

What if I don't need storybook? Sure I can still add those config on the libs but like I said on this thread it only works on `nx serve` not on `nx build`

**Comment by med-albert** [comment]
New Findings, only on `--prod` that PostCSS don't work on libs. Don't have a clue why.

**Comment by danr-za** [comment]
@med-albert - can you share a repo?

**Comment by jaysoo** [comment]
After some discussion with @juristr and @leosvelperez this is a bug. The application config should apply to libraries just as they do for next.js and angular. We'll have this patched up for a future release.

Again, for now use the workaround of adding the tailwind and postcss configs to your lib.
